src/backend/optimizer/util/plancat.c

```diff
@@ -546,8 +555,25 @@ cdb_estimate_rel_size(RelOptInfo   *relOptInfo,
 		int32		tuple_width;
 
 		tuple_width = get_rel_data_width(rel, attr_widths);
-		tuple_width += sizeof(HeapTupleHeaderData);
-		tuple_width += sizeof(ItemPointerData);
+
+		if (RelationIsAoRows(rel))
+		{
+			/*
+			 * Row oriented relations store memtyples, that have 4 byte header,
+			 * plus storage is done in VarBlock, which has item-offsets array
+			 * with at 2 or 3 byte offsets. For simplicity, in our rough
+			 * estimation we consider only 2 byte offsets. Thus, we add 6 bytes
+			 * to the tuple width.
+			 */
+			tuple_width += 6;
```

Comment:
4 is size of uint32(MemTupleData.PRIVATE_mt_len), but compiler can aling a structure fields some other way. So I suggest to use `offsetof`, because it is more reliable
```suggestion
			tuple_width += offsetof(MemTupleData, PRIVATE_mt_bits) + 2;
```